osu.Game/Rulesets/Mods/IMod.cs

```diff
@@ -33,10 +33,24 @@ public interface IMod : IEquatable<IMod>
         /// </summary>
         IconUsage? Icon { get; }
 
+        /// <summary>
+        /// Whether this mod is playable for the given usage.
```

Comment:
This is missing context which results in the purpose of the method being undefined. Is the end purpose of this method in line with the obsoleted one? ie. can we rename it to `RequiresUserInteraction`? Otherwise I'm not sure how a newcomer to the project could make sense of this.

If `IsPlayable(SoloLocal)` is false, a mod is still selectable during solo local, but the mod is non-interactive.

If `IsPlayable(MultiplayerLocal)` (i still cannot get behind this "local" naming, but whatever) is false, the mod should never be shown anywhere.

Is this correct thinking? If so, I have a hard time thinking of how to phrase this xmldoc, but what is currently written definitely doesn't do a good job. At least the obsoleted version mentioned "playable by an end user", which explains what "playable" actually means here.

Reply:
I think the problem here is that the way for checking whether a mod is selectable at multiplayer was fused with `UserPlayable` with the naming, which is completely different from what the method is meant to achieve.

One final suggestion I have is to rename it to `IsValidFor(ModUsage usage)`, or stick back to the old way of having two non-cascading boolean properties (`ValidForMultiplayer`, `ValidForMultiplayerAsFreeMod`) and call it a day.

@bdach need your opinion on this, or if you have a better idea to improve this.

Reply:
I'd prefer two separate `bool`s, personally. If we are sure that these are the two well-defined usages we need. The naming you suggest seems fine.

Reply:
I don't know, I kinda give up. I can live with three boolean flags, so long as they are *separated* from each other and don't fall back to each other via virtual properties like they did before. I hope I'm not the only one who thinks that having three overridable virtual properties check each other in sequence is a guaranteed way to produce some unholy abomination sometime down the line.

Reply:
Is there a usage I'm missing with sequential checks, or are you referring to chaining the values?

The main issue I have is the washy definition of what "playable" is, and how autoplay almost feels like an edge case here.

Reply:
I'm referring to the chaining alone. I don't want to see stuff like this, essentially:

![image](https://user-images.githubusercontent.com/20418176/166912018-692c1d27-77cc-4771-841e-1949a8a65354.png)

which is what this PR initially had.

Reply:
Yeah, can agree with not chaining, don't think there's anything that relies on that anyways.